elastalert/elastalert.py

```diff
@@ -819,11 +820,16 @@ def send_alert(self, matches, rule, alert_time=None):
                 matches[0]['kibana_link'] = kb_link
 
         for enhancement in rule['match_enhancements']:
+            valid_matches = []
             for match in matches:
                 try:
                     enhancement.process(match)
+                    valid_matches.append(match)
+                except DropMatchException as e:
+                    pass
                 except EAException as e:
                     self.handle_error("Error running match enhancement: %s" % (e), {'rule': rule['name']})
+            matches = valid_matches
 
```

Comment:
I think this function should return here if there are no matches left. Calling alerter.alert([]) is *probably* most of the time, but unfortunately I think there are a few places where matches[0] is referenced explicitly. Also, it looks like it would still write back to Elasticsearch that an empty alert was sent.